sdk/iot/provisioning/inc/az_iot_provisioning_client.h

```diff
@@ -77,27 +80,37 @@ AZ_NODISCARD az_result az_iot_provisioning_client_init(
  * @brief Gets the MQTT user name.
  *
  * @param[in] client The #az_iot_provisioning_client to use for this call.
- * @param[in] mqtt_user_name An empty #az_span with sufficient capacity to hold the MQTT user name.
- * @param[out] out_mqtt_user_name The output #az_span containing the MQTT user name.
+ * @param[out] mqtt_user_name A buffer with sufficient capacity to hold the MQTT user name.
+ *                            If successful, contains a null-terminated string with the user name
+ *                            that needs to be passed to the MQTT client.
+ * @param[in] mqtt_user_name_size The size, in bytes of #mqtt_user_name.
+ * @param[out] out_mqtt_user_name_length __[nullable]__ Contains the string length in bytes, of 
+ *                                                      #mqtt_user_name. Can be NULL.
  * @return #az_result
  */
-AZ_NODISCARD az_result az_iot_provisioning_client_user_name_get(
+AZ_NODISCARD az_result az_iot_provisioning_client_get_user_name(
     az_iot_provisioning_client const* client,
-    az_span mqtt_user_name,
-    az_span* out_mqtt_user_name);
+    char* mqtt_user_name,
+    size_t mqtt_user_name_size,
+    size_t* out_mqtt_user_name_length);
 
 /**
  * @brief Gets the MQTT client id.
  *
  * @param[in] client The #az_iot_provisioning_client to use for this call.
- * @param[in] mqtt_client_id An empty #az_span with sufficient capacity to hold the MQTT client id.
- * @param[out] out_mqtt_client_id The output #az_span containing the MQTT client id.
+ * @param[out] mqtt_client_id A buffer with sufficient capacity to hold the MQTT client id.
+ *                            If successful, contains a null-terminated string with the client id
+ *                            that needs to be passed to the MQTT client.
+ * @param[in] mqtt_client_id_size The size, in bytes of #mqtt_client_id.
+ * @param[out] out_mqtt_client_id_length __[nullable]__ Contains the string length in bytes, of
+ *                                                      of #mqtt_client_id. Can be NULL.
  * @return #az_result
  */
-AZ_NODISCARD az_result az_iot_provisioning_client_id_get(
+AZ_NODISCARD az_result az_iot_provisioning_client_get_client_id(
     az_iot_provisioning_client const* client,
-    az_span mqtt_client_id,
-    az_span* out_mqtt_client_id);
+    char* mqtt_client_id,
+    size_t mqtt_client_id_size,

```

Comment:
I thought we were not going to use size_t (use int32_t instead). MIA?

Reply:
What does MIA stand for?

Reply:
We're using size_t. The reason is that char* here is a C string (as defined by the C99 standard - either a literal or the way the std libraries define strings). 
The purpose of the API change is to easily interoperate with C99's `sizeof` and `strlen`.